src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -565,25 +565,29 @@ Namespace Microsoft.CodeAnalysis.Operations
         End Function
 
         Private Function CreateBoundBinaryOperatorOperation(boundBinaryOperator As BoundBinaryOperator) As IBinaryOperation
+            Dim left As IOperation = CreateBoundBinaryOperatorChild(boundBinaryOperator, isLeft:=True)
+            Dim right As IOperation = CreateBoundBinaryOperatorChild(boundBinaryOperator, isLeft:=False)
             Dim binaryOperatorInfo = GetBinaryOperatorInfo(boundBinaryOperator)
             Dim syntax As SyntaxNode = boundBinaryOperator.Syntax
             Dim type As ITypeSymbol = boundBinaryOperator.Type
             Dim constantValue As ConstantValue = boundBinaryOperator.ConstantValueOpt
             Dim isImplicit As Boolean = boundBinaryOperator.WasCompilerGenerated
-            Return New VisualBasicLazyBinaryOperation(Me, boundBinaryOperator, binaryOperatorInfo.OperatorKind, binaryOperatorInfo.IsLifted,
-                                                      binaryOperatorInfo.IsChecked, binaryOperatorInfo.IsCompareText, binaryOperatorInfo.OperatorMethod,
-                                                      unaryOperatorMethod:=Nothing, _semanticModel, syntax, type, constantValue, isImplicit)
+            Return New BinaryOperation(binaryOperatorInfo.OperatorKind, left, right, binaryOperatorInfo.IsLifted,
```

Comment:
The VB handling is rewritten at the end of this PR to be stack-based as well. I didn't want to go and amend this commit in case someone had already started looking, but I'll rebase to squash it in after review.